packages/preview-middleware/src/base/flp.ts

```diff
@@ -475,23 +477,81 @@ export class FlpSandbox {
      * @param manifest manifest of the additional target app
      * @param app configuration for the preview
      */
-    addApp(manifest: Manifest, app: App) {
+    async addApp(manifest: Manifest, app: App) {
         const id = manifest['sap.app'].id;
         app.intent ??= {
             object: id.replace(/\./g, ''),
             action: 'preview'
         };
         this.templateConfig.ui5.resources[id] = app.target;
         this.templateConfig.apps[`${app.intent?.object}-${app.intent?.action}`] = {
-            title: manifest['sap.app'].title ?? id,
-            description: manifest['sap.app'].description ?? '',
+            title: await this.getAppTitle(manifest, app.local),
+            description: await this.getAppDescription(manifest, app.local),
             additionalInformation: `SAPUI5.Component=${app.componentId ?? id}`,
             applicationType: 'URL',
             url: app.target,
             applicationDependencies: { manifest: true }
         };
     }
 
+    /**
+     * Get the i18n properties bundle.
+     *
+     * @param projectRoot absolute path to the project root
+     * @returns i18n properties bundle
+     * @private
+     */
+    private async getPropertiesI18nBundle(projectRoot: string) {
+        //todo: consider relative path from manifest['sap.app'].i18n
+        //todo: consider locale
+        const i18nFilePath = join(projectRoot, 'webapp', 'i18n', 'i18n.properties');
+        let bundle: I18nBundle | undefined;
+        try {
+            bundle = await getPropertiesI18nBundle(i18nFilePath);
+        } catch (e) {
+            this.logger.warn(`Failed to load i18n properties bundle from ${i18nFilePath}`);
+        }
+        return bundle;
+    }
+
+    /**
+     * Get the title of the application from the i18n properties file.
+     *
+     * @param manifest application manifest
+     * @param projectRoot absolute path to the project root
+     * @returns title of the application
+     * @private
+     */
+    private async getAppTitle(manifest: Manifest, projectRoot: string): Promise<string> {
+        const bundle = await this.getPropertiesI18nBundle(projectRoot);
+        const title = manifest['sap.app'].title;
+        const titleI18nKey = title?.substring(2, title.length - 2);
+        let titleText = title ?? manifest['sap.app'].id;
+        if (bundle) {
+            titleText = bundle[titleI18nKey]?.[0]?.['value']?.value ?? titleText;
+        }
+        return titleText;
+    }
+
+    /**
+     * Get the description of the application from the i18n properties file.
+     *
+     * @param manifest application manifest
+     * @param projectRoot absolute path to the project root
+     * @returns description of the application
+     * @private
+     */
+    private async getAppDescription(manifest: Manifest, projectRoot: string): Promise<string> {
```

Comment:
`getAppTitle` and `getAppDescription` look pretty identical. Could we put this together into one function and have `manifest['sap.app'].description` or `manifest['sap.app'].title` as input

Reply:
Good idea 👍🏻 Done